src/main/java/com/liferay/upgrades/analyzer/project/dependency/detector/JSToolKitDetector.java

```diff
@@ -0,0 +1,60 @@
+package com.liferay.upgrades.analyzer.project.dependency.detector;
+
+import com.liferay.upgrades.analyzer.project.dependency.graph.builder.ProjectsDependencyGraphBuilder;
+import com.liferay.upgrades.analyzer.project.dependency.model.ProjectKey;
+
+import java.io.BufferedReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+
+public class JSToolKitDetector implements ProjectDetector {
+
+    @Override
+    public boolean matches(String fileName, Path file) {
+        return fileName.equals("package.json") && _validateJsonContent(file);
+    }
+
+    @Override
+    public void detect(Path file, ProjectsDependencyGraphBuilder projectsDependencyGraphBuilder) {
+        projectsDependencyGraphBuilder.addProject(
+                _getProjectKey(file),  Collections.emptySet());
+    }
+
+    private ProjectKey _getProjectKey(Path file) {
+        String path = file.getParent().toUri().getPath();
+        String fileName = file.getParent().getFileName().toString();
+
+        String key = fileName.trim().replaceAll("'", "").replaceAll("\"", "");
+
+        return new ProjectKey(key, path);
+    }
+
+    private boolean _validateJsonContent(Path contentPath) {
+        String jsonContent = _getContentByPath(contentPath);
+
+        if (jsonContent.isEmpty()) {
+            return false;
+        }
+
+        return jsonContent.contains("portlet");
```

Comment:
This check is weak. It will match everything that contains the word "portlet". Please create a more strong logic detection.